app/src/main/java/com/example/drawingactivity/GradientView.kt

```diff
@@ -0,0 +1,160 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.graphics.Color.BLACK
+import android.graphics.Color.HSVToColor
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import kotlinx.coroutines.*
+
+@SuppressLint("ClickableViewAccessibility")
+class GradientView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onSatValChanged: ((Float, Float) -> Unit)? = null
+    private var hue = 359
+    private var shape: Rect? = null
+    private var dx = (width * 1f - 15f)
+    private var dy = 5f
+    private val selectPaint = Paint().apply {
+        strokeWidth = 3f
+        color = BLACK
```

Comment:
is this easier to see than white?

Reply:
I swap the color between WHITE and BLACK based on where they click